llama_index/llms/huggingface.py

```diff
@@ -474,8 +480,16 @@ def __init__(self, **kwargs: Any) -> None:
         except ModuleNotFoundError as exc:
             raise ImportError(
                 f"{type(self).__name__} requires huggingface_hub with its inference"
-                " extras, please run `pip install huggingface_hub[inference]`."
+                " extra, please run `pip install huggingface_hub[inference]>=0.19.0`."
             ) from exc
+        if kwargs.get("model_name") is None:
+            task = kwargs.get("task", "")  # NOTE: empty string leads to error
```

Comment:
it might help to name some kwargs?

Reply:
Yeah sorry this could have been more clear, I updated the comment to make it more clear.  Fwiw I prefer not to duplicate `Field` names/defaults in `__init__` signature though

Reply:
But then it doesn't show up in the IDE 😢

Reply:
I guess to think in a similar area, IDE autocomplete works for `dataclass`. It works because our IDEs know to look for class-level fields. So really the issue here is IDE not recognizing `pydantic`'s `Field`s for autocomplete.

What is your IDE? Perhaps you can use this IDE plugin: [Pydantic](https://plugins.jetbrains.com/plugin/12861-pydantic)

Tab complete works fine for me with it 👌 :

<img width="1610" alt="screenshot of tab complete" src="https://github.com/run-llama/llama_index/assets/8990777/77243a06-59c9-486f-ad9b-45c620bca6cb">

Reply:
interesting, I could have sworn this didn't work at some point for me with custom  ctors on pydantic objects (especially with attributes on parent classes), but now it seems to magically work? neat